### PR TITLE
correct out of date Java style guide

### PR DESCRIPTION
**Pull request by gracew**

### CODE REVIEW

File: docs/java-style-guide/readme.md [hunk 1]
```diff
@@ -221,12 +221,6 @@ top-level class it contains, plus the `.java` extension.
 
 Source files are encoded in **UTF-8**.
 
-### File length: 2000 lines
```
Comment (gracew) [inline docs/java-style-guide/readme.md:224]: removed in https://github.com/palantir/gradle-baseline/pull/85

File: docs/java-style-guide/readme.md [hunk 2]
```diff
@@ -332,10 +326,8 @@ Unused or redundant imports should be removed.
 
 Static imports of project-specific methods limit readability as they
 obfuscate method provenance. Static imports are thus discouraged with
-the exception of folklore methods under `org.junit.Assert`,
```
Comment (gracew) [inline docs/java-style-guide/readme.md:335]: modified in https://github.com/palantir/gradle-baseline/pull/240

Comment (qinfchen) [reply docs/java-style-guide/readme.md:335]: nit: worth matching https://github.com/palantir/gradle-baseline/blob/219602b5ef1a51bb17b488880696acc7f985dbc1/.baseline/checkstyle/checkstyle.xml#L82-L83, specifically java collector and collection?

Comment (gracew) [reply docs/java-style-guide/readme.md:335]: didn't want to have to update the docs every time we change that checkstyle rule -- but I guess we shouldn't be changing it so much after #240?

**Review by qinfchen — APPROVED** [review]

**Pull request merged**